processor/probabilisticsamplerprocessor/logsprocessor.go

```diff
@@ -75,7 +75,17 @@ func (lsp *logSamplerProcessor) processLogs(ctx context.Context, ld plog.Logs) (
 				if lidBytes == nil && lsp.samplingSource != "" {
 					if value, ok := l.Attributes().Get(lsp.samplingSource); ok {
 						tagPolicyValue = lsp.samplingSource
-						lidBytes = value.Bytes().AsRaw()
+
+						switch value.Type() {
+						case pcommon.ValueTypeStr:
+							lidBytes = []byte(value.Str())
+						case pcommon.ValueTypeBytes:
+							lidBytes = value.Bytes().AsRaw()
+						default:
+							lsp.logger.Warn("incompatible log record attribute, only String or Bytes supported; skipping log record",
```

Comment:
In this case the user has configured a log record attribute that is not a supported type. Unfortunately, the `RemoveIf` functions used by this processor don't return errors, so I figure the only sane thing here without a bigger refactor is to log something. Is `WARN` the appropriate severity for a misconfigured processor?

Reply:
I guess the only value that should not be supported is boolean. Otherwise, the probability provided by the user will be skewed by the fact that booleans can only have two values: there won't be a good distribution of values to make a proper probability.

Reply:
I believe there is a method named, `AsString` which works regardless the type since it just calls `fmt.Sprint` ?

Would that work here instead?